src/modules/form-builder/components/editable-div/editable-div.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormEvent, useEffect, useRef, useState } from "react";
+
+import { cn } from "@/utils/classname";
+
+import { BlockTypesEnum, KeyCodesEnum } from "../../enums/form-builder.enum";
+import { TBlocksStore } from "../../store";
+import { TBlock } from "../../types/form-builder.types";
+import { setBlockInFocus, setCursorToEnd } from "../../utils/form-builder.utils";
+import { FormBuilderMenu } from "../form-builder-menu";
+
+type EditableDivProps = Pick<React.DOMAttributes<HTMLDivElement>, "onKeyDown"> & {
+    className?: string;
+    value: TBlock;
+    totalBlocks: number
+    addBlock?: TBlocksStore["addBlock"];
+    deleteBlock?: TBlocksStore["removeBlock"];
+    onChange?: TBlocksStore["updateBlock"];
+};
+
+export const EditableDiv = ({ value, totalBlocks, addBlock, deleteBlock, className, onChange, onKeyDown }: EditableDivProps) => {
+    const ref = useRef<HTMLDivElement>(null);
+    const [prevKey, setPrevKey] = useState("");
+
+    const innerText = value.payload.data;
+    const showBlockTypeMenu = innerText.length === 1 && innerText[0] === "/";
+
+    const isFormTitle = value.type === BlockTypesEnum.FORM_TITLE;
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+        const currentValue = e.currentTarget.innerText;
+        const isPrevKeyShift = prevKey === KeyCodesEnum.SHIFT;
+
+        const isValueEmpty = !currentValue || currentValue?.length === 0 || currentValue === "\n";
+
+        const elements = document.querySelectorAll("[contenteditable]");
```

Comment:
document.querySelectorAll within handleKeyDown can be inefficient, especially if it's called frequently. Consider caching the element collection or using a more efficient approach for finding the current element's index.

Reply:
- removed this, using the index we get when mapping over all the blocks instead.

#### Test coverage post the change:
<img width="856" alt="image" src="https://github.com/yesyash/wisee-web/assets/56453897/d819f4b9-45a4-4b19-a777-3af14743e309">